pkgs/development/compilers/gerbil/gerbil-support.nix

```diff
@@ -59,18 +118,34 @@ rec {
 
       installPhase = ''
         runHook preInstall
-        mkdir -p $out/gerbil/lib
-        cp -fa .build/lib $out/gerbil/
-        bins=(.build/bin/*)
-        if [ 0 -lt ''${#bins} ] ; then
-          cp -fa .build/bin $out/gerbil/
-          mkdir $out/bin
-          cd $out/bin
-          ln -s ../gerbil/bin/* .
+        mkdir -p $out/gerbil
+        cp -fa .build/* $out/gerbil/
+        if [ -d $out/gerbil/bin ] ; then
+        (cd $out/gerbil
+         bins=`find ../gerbil/bin -type f`
+         if [ -n "$bins" ] ; then
+          (mkdir -p ../bin
+           cd ..
+           ln -s $bins bin)
+           fi)
```

Comment:
```suggestion
        (
          cd $out/gerbil
          bins=$(find ../gerbil/bin -type f)
          if [[ -n $bins ]] ; then
            (
              mkdir -p ../bin
              cd ..
              ln -s $bins bin
            )
          fi
        )
```

Reply:
I see that most files in nixpkgs use `[` rather than `[[`. Is the shell used 100% guaranteed to be bash? In this case, then yes, `[[` might make more sense.

Reply:
I think right now stdenv is guaranteed to use Bash as the shell, but I also agree with you that using Bash-specific version for zero gain in the specifric case is pointless.

Reply:
shell is always bash and [ should always be replaced with [[ because it has less pitfalls.

Reply:
That'll be a whole lot of files to update, and not just among those I maintain. But OK.

Reply:
Done.